server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -290,14 +331,24 @@ public ClusterMetadataManifest writeIncrementalMetadata(
             previousClusterState.metadata(),
             clusterState.metadata()
         ) == false;
-        String globalMetadataFile;
-        // For migration case from codec V0 to V1, we have added null check on global metadata file,
-        // If file is empty and codec is 1 then write global metadata.
-        if (updateGlobalMetadata || previousManifest.getGlobalMetadataFileName() == null) {
```

Comment:
Will we still be able to handle migration from V0 to V1? 
Also, in what cases would we see cluster state with Codec0

Reply:
Currently Codec V0 is associated with version 2.10/2.11 and codec V1 with 2.12. If the customer is upgrading from 2.10/2.11 to 2.13 we support migration to V1, instead we will directly move from V0 to V2.